### ISSUE TITLE
[Suggestion] Gym trainer

### ISSUE THREAD

**Issue by MrJokerer** [jabbink/PokemonGoBot#390]
Idea to make a Gym trainer to help prestige gyms.  
Could be as simple as
-pick highest CP  pokemon,
-train against gym, 
-"tap screen repeatedly", 
-go until fainted
-use hyper/max potion until full *or* pick next highest CP
-repeat

**Comment by alone23** [comment]
+1

**Comment by 2black0** [comment]
+2

**Comment by andredp** [comment]
I don't agree... In my opinion that is messing with other people's fun with the game... It's one thing to have a bot catching pokemons. It's another thing to mess with gyms (even if it's on the same team) and messing with people that are playing the game legit...

It's also a nice way to force Niantic into fixing this whole thing... Imagine creating 100 bots and putting them in a city "conquering" gyms...

**Comment by Agronis** [comment]
I second andredp.. but I feel like Niantic will squash us all the same.

Someone will create a Gym interaction for one of the APIs eventually unfortunately.

At best, a feature to train - not fight.

If Gym == My team -> Train to lvl10.

**Comment by megatixos** [comment]
please make this happen.. :+1:

**Comment by falko100** [comment]
:+1:  +1

**Comment by MrJokerer** [comment]
@andredp while I agree that it's crossing into messing with other people I would like to stress this only for your team.  I understand your point but if we do anything with the pokemon this is catching with gyms we are essentially doing same thing. Was just an idea though.  I don't want to see this be able to take other gyms though.  Idk tricky situation

**Comment by andredp** [comment]
I understand your point about people with botted accounts going into gyms, and it's a fair one. But if people see no one around a gym but the gym is constantly gaining prestige (from multiple bots farming it) even while they're attacking the gym it will piss them off... Especially if a lot of people starts doing it. Then you're breaking ~50% of the game and Niantic is urged to take action.

It's a different case if you bring your bot caught pokemons. You can only up the prestige at a low rate AND you have to be there, so people that don't know you botted will still think it's fair.

I know it's a touchy subject, because of how hard it is to draw the line... Where is it really the limit? I just wanted to expose a different point of view... In the end if it's a popular request it will probably be implemented, if @jabbink agrees.

**Comment by Rofii** [comment]
-1 for this idea
+1 for @andredp's explanation of why not.

**Comment by MrJokerer** [comment]
I understand both sides @andredp thanks for stating the side you did.  Just a suggestion I thought of and in the end its up to @jabbink

**Comment by gabrielklein** [comment]
I think they would start to have more protection on their server if cheaters start to mess with other people.
